src/core/exec-invoke.c

```diff
@@ -5506,10 +5588,27 @@ int exec_invoke(
                         command,
                         needs_sandboxing,
                         have_cap_sys_admin,
+                        bpffs_socket_fd,
                         exit_status);
         if (r < 0)
                 return r;
 
+        if (context->private_bpf != PRIVATE_BPF_NO) {
+                r = pidref_wait_for_terminate_and_check("(sd-bpffs)", &bpffs_pid, /* flags = */ 0);
+                TAKE_PIDREF(bpffs_pid);
```

Comment:
This should be moved after the error handling and be replaced with `pidref_done()`, otherwise we'll leak the pidref